easybuild/tools/modules.py

```diff
@@ -557,23 +557,32 @@ def mod_exists_via_show(mod_name):
             mod_exists_regex = r'\s*/.+:\s*'
             for line in stderr.split('\n'):
                 # skip whitespace lines
-                self.log.debug("Checking line '%s' to determine whether %s exists...", line)
+                self.log.debug("Checking line '%s' to determine whether %s exists...", line, mod_name)
                 if OUTPUT_MATCHES['whitespace'].search(line):
                     self.log.debug("Treating line '%s' as whitespace, so skipping it", line)
                     continue
 
                 # if any errors occured, conclude that module doesn't exist
                 if OUTPUT_MATCHES['error'].search(line):
+                    self.log.debug("Line '%s' looks like an error, so concluding that %s doesn't exist", mod_name)
                     break
 
+                # skip warning lines, which may be produced by modules tool but should not be used
+                # to determine whether a module file exists
+                if line.startswith('WARNING: '):
```

Comment:
Do you have a use case and sample output for this? Seems to haven't come up yet.

Reply:
@casparvl Shared `module show Java/1.8.0_192` stderr output with me that starts with:

```
module-version    Java/1.8.0_192 1.8
WARNING: Duplicate version symbol '1.8' found
module-version  Java/1.8.0_192 1.8
-------------------------------------------------------------------
/sw/arch/Debian10/EB_production/2019/modulefiles/lang/Java/1.8.0_192:
```

Reply:
Hm... What happens if they are different versions? Shall we really skip those?

Reply:
Also mentioned this in #3376, and I'll add a test case for this...

Reply:
I know the PR has already been merged, but just to log this: I don't think it is an issue if the versions are different.

I hit this issue when trying to install the wrapper (i.e. `Java-1.8`). The EasyConfig specifies the _specific_ version (here: `1.8.0_192`) that you want to wrap. The log show that in the sanity check, the `module show` command is called on that _specific_ version (`Checking whether Java/1.8.0_192 exists based on output of 'module show'`). That will just return the correct output (i.e. it will either find a module, or not), regardless of what is set in the `.modulerc`. (since the module show is called on an actual module, and not the wrapper, I'm not even entirely sure why a `module show 1.8.0_192` even involves reading the two `.modulerc` files, but who cares)

Note that if any EasyBuild command would ever do a `module show Java/1.8` to check if thát module exists, I agree that it would be a concern to check which underlying specific module is actually checked for. I tried `module show Java/1.8`, and it just picks up on the `.modulerc` that comes first in the `MODULEPATH`. Of course, it _could_ happen that the `.modulerc` that comes first in the `MODULEPATH` is not the the one that is in EasyBuilds `installpath-modules`. I guess you could argue if that is the desired behaviour, or if you want EasyBuild to check based on the version defined in the `.modulerc` that IS in the `installpath-modules`. However, that nuance true in general for any check that would rely on `module show` - not just for modules that are wrapped - and I therefore don't think we should try to resolve that here.